walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,80 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays; 
+
 
 public final class FindMeetingQuery {
+
+  /**
+   * Returns {@code TimeRange} when a meeting can happen based on other events and the
+   * meeting participants involved.
+   * @param events {@code Collection} of events happening.
+   * @param request {@code MeetingRequest} for a meeting
+   * @return Collection of {@code TimeRange} when a meeting can happen. 
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    final int MINUTES_IN_A_DAY = 1440;
+    ArrayList<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+    ArrayList<Event> localEvents = new ArrayList<Event>(events);
+    boolean[] availableTimes = new boolean[MINUTES_IN_A_DAY+1];
```

Comment:
Does `availableTimes` actually end up being an array where all available times are marked with `true`?

I'm not certain that's true and I think it's a core piece of understanding what the algorithm is doing.

Reply:
I was actually marking the busy times. It was available times at first but then I marked the times with meetings.
Will rename accordingly.